generator/src/test/java/com/graphicsfuzz/generator/transformation/donation/CheckDonorVersionMatchesTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.generator.transformation.donation;
+
+import com.graphicsfuzz.common.ast.TranslationUnit;
+import com.graphicsfuzz.common.transformreduce.GlslShaderJob;
+import com.graphicsfuzz.common.transformreduce.ShaderJob;
+import com.graphicsfuzz.common.util.ParseHelper;
+import com.graphicsfuzz.common.util.PipelineInfo;
+import com.graphicsfuzz.common.util.RandomWrapper;
+import com.graphicsfuzz.common.util.ShaderJobFileOperations;
+import com.graphicsfuzz.common.util.ShaderKind;
+import com.graphicsfuzz.generator.transformation.DonateLiveCodeTransformation;
+import com.graphicsfuzz.generator.util.GenerationParams;
+import com.graphicsfuzz.generator.util.TransformationProbabilities;
+import java.io.File;
+import java.util.Optional;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class CheckDonorVersionMatchesTest {
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Test(expected = RuntimeException.class)
```

Comment:
As "RuntimeException" is a very general class, there is a danger that this test will pass if some unrelated RuntimeException is thrown.

I suggest instead wrapping the code that is expected to throw an exception in a "try ... catch" as follows:

try {
  callToFunction(); // call that should throw a RuntimeException
  fail("An exception should have been thrown.");
} catch (RuntimeException runtimeException) {
  assertTrue(runtimeException.startsWith("Incompatible versions");
}

Reply:
done